src/plays/random-meme-generator/RandomMemeGenerator.jsx

```diff
@@ -0,0 +1,73 @@
+
+import { useLocation } from 'react-router-dom';
+import { getPlayById } from 'meta/play-meta-util';
+import { useState, useEffect } from 'react';
+
+import PlayHeader from 'common/playlists/PlayHeader';
+import './random-meme-generator.css';
+
+function RandomMemeGenerator() {
+  // Do not remove the below lines. 
+  // The following code is to fetch the current play from the URL
+  const location = useLocation();
+  const { id } = location.state;
+  const play = getPlayById(id);
+
+  // Your Code Start below.
+
+  const [meme, setMeme] = useState({});
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  function getMeme() {
+    setIsLoading(true);
+    fetch('https://meme-api.herokuapp.com/gimme').then((res) => {
+      return res.json();
+    }).then((data) => {
+      setIsLoading(false)
+      if(data.nsfw) {
+        getMeme();
+      } else {
+        console.log(data);
+        setMeme((prev) => {
+          return data;
+        })
+      }
+    }).catch((err) => {
+      console.error(err);
+    })
+  }
+
+  useEffect(() => {
+    getMeme();
+  }, [])
+
+  function handleClick(e) {
+    e.target.classList.contains('generate-btn') && getMeme();
```

Comment:
Why is this check necessary?

Reply:
Removed it now, actually I was using 2 buttons back then, but now the function isn't needed anymore.